api/internal/localizer/localizer.go

```diff
@@ -127,3 +134,43 @@ func (lc *Localizer) localizeFile(path string) (string, error) {
 	}
 	return locPath, nil
 }
+
+// localizeDir localizes root path and returns the localized path
+func (lc *Localizer) localizeDir(path string) (string, error) {
+	ldr, err := lc.ldr.New(path)
+	if err != nil {
+		return "", errors.Wrap(err)
+	}
+	defer func() { _ = ldr.Cleanup() }()
+
+	var locPath string
+	if repo := ldr.Repo(); repo != "" {
+		// TODO(annasong): check if able to add localize directory
+		locPath = locRootPath(path, filesys.ConfirmedDir(repo), filesys.ConfirmedDir(ldr.Root()))
+	} else {
+		locPath, err = filepath.Rel(lc.ldr.Root(), ldr.Root())
+		if err != nil {
+			// This should never occur, but if hit, something is fundamentally wrong and we should immediately exit.
+			//nolint:gocritic
+			log.Fatalf("cannot find relative path between properly scoped localize roots %q and %q: %s",
+				lc.ldr.Root(), ldr.Root(), err.Error())
+		}
+		// TODO(annasong): check if locPath conflicts with localize directory
+	}
+	newDst := lc.dst.Join(locPath)
+	if err = lc.fSys.MkdirAll(newDst); err != nil {
+		return "", errors.WrapPrefixf(err, "unable to create root %q in localize destination", path)
+	}
+	err = (&Localizer{
+		fSys:      lc.fSys,
+		validator: lc.validator,
+		rFactory:  lc.rFactory,
+		pLdr:      lc.pLdr,
+		ldr:       ldr,
+		dst:       filesys.ConfirmedDir(newDst),
```

Comment:
similar comment here: should we be actually confirming this dir rather only declaring it to have been confirmed?

Reply:
Will do.

Reply:
I don't think we need to call `filesys.ConfirmDir()` in this case because the previous `dst` is a `filesys.ConfirmedDir`, `toDst` is clean, and we call `MkdirAll` to create `toDst`, which, as a result, cannot contain symlinks. 

I changed all other instances of the conversion that actually assumed `Root()` and `Repo()` returned `ConfirmedDir`s, though.

Reply:
There shouldn't be any symlinks because `newDst` must be in the localize destination directory, which `LocLoader` created. 

I like the suggestion, but I know it's a redundant check. I'm thinking I'll change the structure of the `Localizer` code so that the destination directory is never exposed. Then, we can be sure that it doesn't contain any symlinks.

Reply:
Should be addressed.

Reply:
@KnVerey ready for review

Following our very helpful discussion today, I realized that `dst` (contrary to some other values) doesn't actually need to be a `filesys.ConfirmedDir`. I changed the `filesys.ConfirmedDir` to `string`.